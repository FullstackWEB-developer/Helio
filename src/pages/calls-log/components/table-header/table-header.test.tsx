import i18n from '../../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import TableHeader from './table-header';

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("Table Header tests", () => {
    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                
            }
        },
        lookupsState: {
            userList: [{
                id: "123",
                firstName: "Emre",
                lastName: "Hanci",
            }]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        callsLogState: {
            isFiltered: false
        },
        ticketState: {
            lookupValues: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders table-header correctly", async () => {
        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <TableHeader>
                <h2>Test</h2>
            </TableHeader>
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });
})