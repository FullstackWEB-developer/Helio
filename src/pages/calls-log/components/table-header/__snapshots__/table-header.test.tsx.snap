// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Header tests renders table-header correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row items-center content-center h-12 bg-gray-100 body2-medium"
  >
    <h2>
      Test
    </h2>
  </div>
</DocumentFragment>
`;
